pkg/operation/botanist/component/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh

```diff
@@ -88,11 +94,14 @@ users:
 - name: kubelet-bootstrap
   user:
     as-user-extra: {}
-    token: {{ .bootstrapToken }}
+    token: "$BOOTSTRAP_TOKEN"
 EOF
 
 else
   rm -f "{{ .pathKubeletKubeconfigBootstrap }}"
+  if [[ -f "{{ .pathBootstrapToken }}" ]]; then
```

Comment:
The `if` should not be needed here, `rm -f` handles this automatically

Reply:
you are right. It exits with 0. Thanks, didn't know that.